sklearn/linear_model/_logistic.py

```diff
@@ -665,7 +665,7 @@ def _logistic_regression_path(X, y, pos_class=None, Cs=10, fit_intercept=True,
     if isinstance(class_weight, dict) or multi_class == 'multinomial':
         class_weight_ = compute_class_weight(class_weight,
                                              classes=classes, y=y)
-        sample_weight *= class_weight_[le.fit_transform(y)]
+        sample_weight = sample_weight * class_weight_[le.fit_transform(y)]
```

Comment:
@rth instead of doing this, would it be better to have `copy=False/True` within `_check_sample_weight` signature?